client/src/utils/view-provider.tsx

```diff
@@ -0,0 +1,146 @@
+    const [completedStep, setCompletedStep] = useState<{
+		[k: number]: boolean;
+	}>({});
+    const handleStep = (step: number) => () => {
+		setActiveStep(step);
+	};
+    interface IActiveKeys {
+        [key: number]: string;
+    }
+    const activeKeys: IActiveKeys = {
+        0: TELEMETRY_KEY,
+        1: FLIGHT_KEY,
+        2: RECOVERY_KEY,
+        3: FLIGHT_REPORT_KEY
+    }
+    const activeStepKeys = ['Start up', 'Flight', 'Recovery', 'Flight Report'];
+	
+    const [isSettingsOpen, setIsSettingsOpen] = useState(false);
+    
+   
+	const handleSettingsDialog = () => {
+		setIsSettingsOpen(!isSettingsOpen);
+	};
+
+    function currentView(viewKey: string) {
+        switch (viewKey) {
+            case ROCKET_SELECT_KEY:
+                return <RocketSelectionView setCurrentView={updateView} />;
+            case COMPONENT_DOCUMENT_KEY:
+                return <ComponentDocs />;
+            case TELEMETRY_KEY:
+                return <TelemetryView />;
+            case FLIGHT_KEY:
+                return <ModulesView />;
+            default:
+                return <RocketSelectionView setCurrentView={updateView} />;
+        }   
+    }
+
+    useEffect(() => {
+        console.log(currentViewKey);
+    }, [currentViewKey])
+    return (
+        <div className='app'>
+            {currentViewKey !== (TELEMETRY_KEY || FLIGHT_KEY || RECOVERY_KEY || FLIGHT_REPORT_KEY)  && currentView(ROCKET_SELECT_KEY)}
+            {currentViewKey === (TELEMETRY_KEY || FLIGHT_KEY || RECOVERY_KEY || FLIGHT_REPORT_KEY)  && (
+				<Grid container spacing={2} direction="row">
+					{/* Any views should be rendered within this grid item */}
+					<Grid item xs={10}>
+						{currentView(activeKeys[activeStep])}
+					</Grid>
+
+					<Grid item xs={2}>
+						<Grid
+							paddingX="1rem"
+							paddingY="1rem"
+							container
+							direction="column"
+							justifyContent="space-between"
+							height="100%"
+							style={fullHeight}
+						>
+							{/* TODO: Should call a Setting pop up */}
+							<Grid item>
+								<SettingsDialog isOpen={isSettingsOpen} onClose={() => setIsSettingsOpen(false)} />
+								<Button variant="outlined" onClick={() => handleSettingsDialog()}>
+									Settings
+								</Button>
+							</Grid>
+
+							{/* Page change stepper */}
+							<Grid container justifyContent="center">
+								<Stepper nonLinear activeStep={activeStep} orientation="vertical">
+									{activeStepKeys.map((label, index) => (
+										<Step key={label} completed={completedStep[index]}>
+											<StepButton color="inherit" onClick={handleStep(index)}>
+												{label}
+											</StepButton>
+										</Step>
+									))}
+								</Stepper>
+							</Grid>
+
+							{/* TODO: Should terminate all data readings */}
+							<Grid item>
+								<Button
+									fullWidth={true}
+									variant="contained"
+									color="error"
+									onClick={() => setCurrentViewKey(ROCKET_SELECT_KEY)}
```

Comment:
After ending the mission, it should head to the flight report screen not selection screen. (Shouldn't be able to download flight report in the middle of data collection.)